### DISCUSSION TITLE
How to run Nextjs behind an Apache proxy?

### DISCUSSION THREAD

**Discussion by dtkerns** [vercel/next.js#52426] [Help] [answered]
### Summary

I've done a lot of (Internet) searching as well as reading through these discussions and the Nextjs docs (I find nothing useful there on proxy) but I'm left wanting for what I think should be an easy config. #10935 seems to hit pretty close to what I want, but even that issue is not resolved/answered. (I have no qualms about using http between Apache and the Nextjs server)

I have a stand-alone server running Nextjs on port 8060 for development; everything works great. Now I want to deploy it behind an (existing) HTTPS Apache server using certificate based authentication hosting many other services. I thought it would be as easy as setting up a proxy path in my Apache conf file: (as I've done for many other services)

```
ProxyPass /mynxjsproj http://localhost:8060
ProxyPassReverse /mynxjsproj http://localhost:8060
```

but it turns out Nextjs uses full paths (starts with /) for everything which totally breaks the proxy. Since this is our first Nextjs app, I added a second proxy for:

```
ProxyPass /_next http://localhost:8060
ProxyPassReverse /_next http://localhost:8060
```

and then I made a few edits to one of .ts files to remove leading / and have gotten a few pages to successfully get served through the proxy, but when I globally remove all leading / they "magically" get put back in during the build process. Also, if we were to deploy a second Nextjs app on the same server, the proxy for `/_next` would become an issue. If Nextjs used relative paths for everything I'm pretty sure it would just work.

Most notably: /api is already in use by another proxy service. Any/All help appreciated.

**Comment by dtkerns** [comment, marked as answer]
I hacked a workable solution... 

in next.config.js I added:

```diff
 const nextConfig = {
   reactStrictMode: true,
+  basePath: '/mynxjsproj',
+  assestPrefix: '/mynxjsproj',
  compiler: {
   styledComponents: true
  }
```

Then in any of the .tx/.tsx files where it did a `fetch()` or `href=""` I added the same prefix.
This allowed the proxy (in the above OP, minus the proxy for /_next) to make it all work.

**Reply by DiskCrasher** [reply]
@dtkerns My Apache proxy config is similar to yours but setting both `basePath` and `assetPrefix` didn't work for me, at least not when trying to use the Next.js [Link](https://nextjs.org/docs/app/api-reference/components/link) component. Works just fine when I use plain old HTML `<a href={'${prefix}/mypath'}>`. Also had similar problems with the [Image](https://nextjs.org/docs/app/api-reference/components/image) component. This is unfortunate because those components should ideally be used in a Next.js application but attempting to get them working with Apache proxy has been a nightmare and ultimately ended in failure. Can't find much info on the Internet with regards to this particular configuration either.
